core/arch/arm/plat-k3/drivers/sa2ul.c

```diff
@@ -28,9 +28,10 @@
 #define FW_ENABLE_REGION        0x0a
 #define FW_BACKGROUND_REGION    BIT(8)
 #define FW_BIG_ARM_PRIVID       0x01
+#define FW_TIFS_PRIVID          0xca
 #define FW_WILDCARD_PRIVID      0xc3
-#define FW_SECURE_ONLY          GENMASK_32(8, 0)
-#define FW_NON_SECURE           GENMASK_32(16, 0)
+#define FW_SECURE_ONLY          GENMASK_32(7, 0)
+#define FW_NON_SECURE           GENMASK_32(15, 0)
```

Comment:
Hi Kamlesh, I see that although both the fixes are related but are still not apt to come in a same commit. Probably split this fix out into a separate commit.